### ISSUE TITLE
Enable MultiRun (-M) as default behaviour

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#479]
Version 5.0.0 may be a good point to change the default behaviour about filenames of temporary files.

See the discussions in #112 and #284.

Options:

0. Don't change the behaviour.
1. Always use the multi-run mode. Retain `-M` for compatibility, but it does nothing.
2. Change the default behaviour to enable the multi-run mode, disabled by, say, `-nM`.

Related, but another concern is that `((int)GetPID())%100000` does not 100% guarantee a unique number. We may change this, for example, by removing `%100000`. Nowadays, I see PIDs like 25356 and 128171. It might happen the situation with PIDs differ by exactly 100000.

Any thoughts?

**Comment by jodavies** [comment]
I think 1 is the easiest to implement, and we could do it for v5 (and also 4.3.2?) since it makes FORM a little more robust and there is no real reason why the user should have "semi" control over the temp filenames. Then the -M manual entry can be updated to say that it does nothing.

At the same time we can use a few more digits of the PID: then one also has to make the (fixed) length of the filenames correspondingly longer.

**Comment by jodavies** [comment]
Something like https://github.com/jodavies/form/tree/multirun ?

This uses 7 digits of PID (on my system at least, the max PID is 4194304 as given by `/proc/sys/kernel/pid_max`. There was also a bug introduced by the form5 merge which meant that the thread sort files never had "filenum" appended, but rather ended just in ".".

The size of the filename buffer was different in different places, but this never realistically caused an issue because the buffer was set to be least 200 chars.

**Comment by tueda** [comment]
I think it is good to implement this change for v5 (and probably also for v4.3.2), but

- In the 2nd commit of the `multirun` branch, `len` can be computable, e.g., `len = t - 1 - sort->file.name`. `(i+DEFAULTFNAMELENGTH)` should be `(i+1)`/`(i+12)`; here 1 is for a null terminator "\0", 12 is for ".%d\0" with an `int` value (assumed to be 32-bit and positive, 10 characters at most).

- The manpage file should also be modified for the `-M` option.

- While you're at it, can you make a commit to replace `':'` with `PATHSEPARATOR` in `ReserveTempFiles()`?

**Comment by jodavies** [comment]
You are right, I missed the reason why the computed buffer space did not work. It simply was using the wrong buffer.

Indeed `i` already contains the filename length. I put the +1 and +12 back and added comments explaining the "magic" numbers.

Manpage modified also, and `:` replaced.

I'll make this a PR for any further discussion.

**Comment by jodavies** [comment]
-M is now the default for v5 (#591)
